pycbc/inference/sampler/dynesty.py

```diff
@@ -0,0 +1,270 @@
+        dlogz = float(cp.get(section, "dlogz"))
+        loglikelihood_function = \
+            get_optional_arg_from_config(cp, section, 'loglikelihood-function')
+        # get constraints since we can't use the joint prior distribution
+        constraints = read_constraints_from_config(cp)
+        obj = cls(model, nlive=nlive, dlogz=dlogz, nprocesses=nprocesses,
+                  loglikelihood_function=loglikelihood_function,
+                  use_mpi=use_mpi)
+        return obj
+
+    def checkpoint(self):
+        pass
+
+    def finalize(self):
+        logz = self._sampler.results.logz[-1:][0]
+        dlogz = self._sampler.results.logzerr[-1:][0]
+        logging.info("log Z, dlog Z: {}, {}".format(logz, dlogz))
+        for fn in [self.checkpoint_file]:
+            with self.io(fn, "a") as fp:
+                fp.write_logevidence(logz, dlogz)
+        logging.info("Writing samples to files")
+        for fn in [self.checkpoint_file, self.backup_file]:
+            self.write_results(fn)
+        logging.info("Validating checkpoint and backup files")
+        checkpoint_valid = validate_checkpoint_files(
+            self.checkpoint_file, self.backup_file)
+        if not checkpoint_valid:
+            raise IOError("error writing to checkpoint file")
+
+    @property
+    def model_stats(self):
+        logl = self._sampler.results.logl
+        return {'loglikelihood': logl}
+
+    @property
+    def samples(self):
+        samples_dict = {p: self.posterior_samples[:, i] for p, i in
+                        zip(self.model.variable_params, range(self.ndim))}
+        return samples_dict
+
+    def set_initial_conditions(self, initial_distribution=None,
+                               samples_file=None):
+        """Sets up the starting point for the sampler.
+
+        Should also set the sampler's random state.
+        """
+        pass
+
+    def write_results(self, filename):
+        """Writes samples, model stats, acceptance fraction, and random state
+        to the given file.
+
+        Parameters
+        -----------
+        filename : str
+            The file to write to. The file is opened using the ``io`` class
+            in an an append state.
+        """
+        with self.io(filename, 'a') as fp:
+            # write samples
+            fp.write_samples(self.samples, self.model.variable_params)
+            # write stats
+            fp.write_samples(self.model_stats)
+            # write log evidence
+            fp.write_logevidence(self._sampler.results.logz[-1:][0],
+                                 self._sampler.results.logzerr[-1:][0])
+
+    @property
+    def posterior_samples(self):
+        """
+        Returns posterior samples from nested samples and weights
+        given by dynsety sampler
+        """
+
+        dynesty_samples = self._sampler.results['samples']
+        weights = numpy.exp(self._sampler.results['logwt'] - \
+                  self._sampler.results['logz'][-1])
```

Comment:
You don't need the line continuation `\` since it's inside parentheses. Also, I think according to PEP8, the `self._sampler.results` on the next line needs to be indented so that it lines up with the `(` on the previous line.